kustomize/commands/edit/fix/convert_test.go

```diff
@@ -1143,3 +1143,76 @@ metadata:
     a.b.c: SOME_SECRET_NAME_PLACEHOLDER
 `, string(content))
 }
+
+func TestFixVarsWithPatch(t *testing.T) {
+	kustomization := []byte(`
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+patchesStrategicMerge:
+  - patch.yaml
+
+vars:
+  - name: CERTIFICATE_NAMESPACE
+    objref:
+      name: system
+    fieldref:
+      fieldpath: metadata.namespace
+`)
+	patch := []byte(`
+apiVersion: apps/v1
+kapiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+        - name: $(CERTIFICATE_NAMESPACE)
+`)
+
+	fSys := filesys.MakeFsInMemory()
+	testutils_test.WriteTestKustomizationWith(fSys, kustomization)
+	assert.NoError(t, fSys.WriteFile("patch.yaml", patch))
+	cmd := NewCmdFix(fSys, os.Stdout)
+	assert.NoError(t, cmd.Flags().Set("vars", "true"))
+	assert.NoError(t, cmd.RunE(cmd, nil))
+	content, err := testutils_test.ReadTestKustomization(fSys)
+	assert.NoError(t, err)
+
+	assert.Equal(t, `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+
```

Comment:
Are these empty lines meant to be here?

Reply:
Yes, as there is an empty line between L1150 and L1152

Reply:
I see, makes sense!